packages/tools/examples/segmentBidirectionalTool/index.ts

```diff
@@ -0,0 +1,435 @@
+  brushInstanceNames.ThresholdBrush,
+];
+
+const optionsValues = [
+  ...brushValues,
+  RectangleScissorsTool.toolName,
+  CircleScissorsTool.toolName,
+  SphereScissorsTool.toolName,
+  PaintFillTool.toolName,
+  BidirectionalTool.toolName,
+];
+
+// ============================= //
+addDropdownToToolbar({
+  options: { values: optionsValues, defaultValue: BrushTool.toolName },
+  onSelectedValueChange: (nameAsStringOrNumber) => {
+    const name = String(nameAsStringOrNumber);
+    const toolGroup = ToolGroupManager.getToolGroup(toolGroupId);
+
+    // Set the currently active tool disabled
+    const toolName = toolGroup.getActivePrimaryMouseButtonTool();
+
+    if (toolName) {
+      toolGroup.setToolDisabled(toolName);
+    }
+
+    if (brushValues.includes(name)) {
+      toolGroup.setToolActive(name, {
+        bindings: [{ mouseButton: MouseBindings.Primary }],
+      });
+    } else {
+      const toolName = name;
+
+      toolGroup.setToolActive(toolName, {
+        bindings: [{ mouseButton: MouseBindings.Primary }],
+      });
+    }
+  },
+});
+
+addSliderToToolbar({
+  title: 'Brush Size',
+  range: [5, 50],
+  defaultValue: 25,
+  onSelectedValueChange: (valueAsStringOrNumber) => {
+    const value = Number(valueAsStringOrNumber);
+    segmentationUtils.setBrushSizeForToolGroup(toolGroupId, value);
+  },
+});
+
+addDropdownToToolbar({
+  options: { values: ['1', '2', '3'], defaultValue: '1' },
+  labelText: 'Segment',
+  onSelectedValueChange: (segmentIndex) => {
+    segmentation.segmentIndex.setActiveSegmentIndex(
+      segmentationId,
+      Number(segmentIndex)
+    );
+  },
+});
+
+addButtonToToolbar({
+  title: 'Find Bidirectional',
+  onClick: () => {
+    const segmentationsList = segmentation.state.getSegmentations();
+    const segmentIndex =
+      segmentation.segmentIndex.getActiveSegmentIndex(segmentationId);
+    const config1 = segmentation.config.getSegmentSpecificConfig(
+      toolGroupId,
+      representationUID,
+      segmentIndex
+    ) || { label: `Segment ${segmentIndex}`, color: null };
+    const segments = [null];
+    segments[segmentIndex] = config1;
+    const segmentations = {
+      ...segmentationsList[0],
+      segments,
+    };
+    const bidirectional = contourAndFindLargestBidirectional(segmentations);
+
+    if (bidirectional) {
+      const { handle0, handle1, handle2, handle3, maxMajor, maxMinor } =
+        bidirectional;
+
+      instructions.innerText = `
+    Major Axis: ${handle0}-${handle1} length ${roundNumber(maxMajor)}
+    Minor Axis: ${handle2}-${handle3} length ${roundNumber(maxMinor)}
+    `;
+
+      const viewport = renderingEngine.getViewport(viewportId1);
+      const bidirectionalToolData = createBidirectionalToolData(
+        bidirectional,
+        viewport
+      );
+      const { FrameOfReferenceUID, referencedImageId } = bidirectional;
+      annotation.state.addAnnotation(
+        bidirectionalToolData,
+        FrameOfReferenceUID
+      );
```

Comment:
I understand that we don't want to include this in the library, but I believe that the contourAndFindLargestBidirectional function is actually a helpful utility. Instead of being an example, it could be categorized as a tools/utility/measurement or something internal.

Reply:
I wasn't sure where to put it initially, so I figured it would move once we decided where it goes.  Maybe adapters is not a bad place to put it actually - alongside the other code?

Reply:
Read my main comment, I think this is a tool